### PR TITLE
Initial front-end setup

### PR DESCRIPTION
**Pull request by giuband**
Just setting up a state of art FE environment for our dear Freesound. 😜 
Run `npm install` and `npm run styleguide` to see the docs.

### CODE REVIEW

File: freesound/static/webpack/common.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
+const path = require('path');
+
+const rootPath = path.resolve('.');
+
+const SUPPORTED_BROWSERS = [
```
Comment (giuband) [inline freesound/static/webpack/common.js]: This is probably the most important line in the entire PR @ffont @alastair

Comment (giuband) [reply freesound/static/webpack/common.js]: It sets up both css autoprefixing and js transpiling according to this setting

File: package.json [hunk 2]
```diff
@@ -1,54 +1,54 @@
...
+    "eslint-config-prettier": "^2.9.0",
+    "eslint-plugin-import": "^2.8.0",
+    "extract-text-webpack-plugin": "^3.0.2",
+    "html-webpack-plugin": "^2.30.1",
+    "node-sass": "^4.7.2",
+    "postcss-flexbugs-fixes": "^3.2.0",
+    "postcss-loader": "^2.0.9",
+    "prettier": "^1.8.2",
+    "react": "^16.2.0",
+    "react-dom": "^16.2.0",
```
Comment (giuband) [inline package.json]: `react` and `react-dom` have been added only to have the `styleguide` correctly running

File: freesound/static/webpack/common.js (hunk 1 above)
Comment (ffont) [reply freesound/static/webpack/common.js]: I think we said we can drop IE < 10 (or even < 11) so I think this list is fine. In any case if this thing lets us know when things won't be supported in non-supported browsers (like with a warning or something), then we'll be better able to decide whether we want to have a fallback implementation of something or just leave it totally unsupported. For example, maybe we have a super basic version of the player for old browsers so these can still be used for basic navigation and for playing sounds.

**Comment by ffont** [comment]
The style guide looks awesome :)
@giuband Do you want me merge this PR?
If you plan on continue working on it before "requesting" a merge, you can add the *[WIP]* prefix to the title (see https://github.com/MTG/freesound/pull/934 for example).

**Comment by giuband** [comment]
I would say let's merge it for now. I'll keep opening more and more for each chunk of work.

**Pull request merged**